site/app/models/GradeableTestcase.php

```diff
@@ -150,6 +164,7 @@ public function getAutochecks() {
     }
     
     public function hasDetails() {
+        if ($this->isHidden()) return false;
         return count($this->autochecks) > 0 || $this->hasCompilationOutput() || $this->hasExecuteLog();
```

Comment:
I think it'd be better to just make this:
```
return (count($this->autochecks) > 0) || $this->hasCompilationOutput() || $this->hasExecuteLog()) && !$this->isHidden()
```

I really dislike one line if statements